lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
                 $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
             }
 
+            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
+             * - the 'client_encoding' connection param only works with postgres >= 9.1
+             * - passing client_encoding via the 'options' param breaks pgbouncer support
+             */
+            if (isset($params['charset'])) {
+              $pdo->query('SET NAMES \''.$params['charset'].'\'');
```

Comment:
My phpinfo ![img](https://cloud.githubusercontent.com/assets/5285820/6972858/d23f74c4-d986-11e4-8267-6a2fc26776b1.jpg)
and server postgres 9.3.4 is on another server

With
$dsn .= 'client_encoding=' . $params['charset'] . ' ';
i confirm that don't work.